sycl/unittests/assert/assert.cpp

```diff
@@ -319,6 +319,20 @@ static ur_result_t redefinedProgramGetInfo(void *pParams) {
     return UR_RESULT_SUCCESS;
   }
 
+  // Required if program cache eviction is enabled.
+  if (UR_PROGRAM_INFO_BINARY_SIZES == *params.ppropName ||
+      UR_PROGRAM_INFO_NUM_DEVICES == *params.ppropName) {
```

Comment:
```suggestion
  if (UR_PROGRAM_INFO_BINARY_SIZES == *params.ppropName) {
```
`UR_PROGRAM_INFO_NUM_DEVICES` case is already defined further up.